tests/Adapter/InMemoryItemsTest.php

```diff
@@ -0,0 +1,395 @@
+<?php
+
+declare(strict_types=1);
+
+namespace tests\Libero\ContentApiBundle\Adapter;
+
+use Libero\ContentApiBundle\Adapter\InMemoryItems;
+use Libero\ContentApiBundle\Exception\ItemNotFound;
+use Libero\ContentApiBundle\Exception\UnexpectedVersionNumber;
+use Libero\ContentApiBundle\Exception\VersionNotFound;
+use Libero\ContentApiBundle\Model\ItemId;
+use Libero\ContentApiBundle\Model\ItemList;
+use Libero\ContentApiBundle\Model\Items;
+use Libero\ContentApiBundle\Model\ItemVersion;
+use Libero\ContentApiBundle\Model\ItemVersionNumber;
+use PHPUnit\Framework\TestCase;
+use function iterator_to_array;
+use function tests\Libero\ContentApiBundle\stream_from_string;
+
+final class InMemoryItemsTest extends TestCase
+{
+    /**
+     * @test
+     */
+    public function it_is_items() : void
+    {
+        $items = new InMemoryItems();
+
+        $this->assertInstanceOf(Items::class, $items);
+    }
+
+    /**
+     * @test
+     */
+    public function it_adds_and_removes_items() : void
+    {
+        $items = new InMemoryItems();
+
+        $item1Id = ItemId::fromString('foo');
+        $item2Id = ItemId::fromString('bar');
+
+        $item1 = new ItemVersion(
+            $item1Id,
+            ItemVersionNumber::fromInt(1),
+            stream_from_string('foo'),
+            'foo'
+        );
+
+        $item2 = new ItemVersion(
+            $item2Id,
+            ItemVersionNumber::fromInt(1),
+            stream_from_string('bar'),
+            'bar'
+        );
+
+        $items->add($item1);
+        $items->add($item2);
+
+        $this->assertCount(2, $items);
+        $this->assertEquals([$item1, $item2], iterator_to_array($items));
```

Comment:
While looking at other implementations (filesystem and DB), this doesn't work. Turns out it only works here as the resource is the same instance as the one provided on input. https://github.com/sebastianbergmann/comparator/blob/71bdaddd52ddb753a144c78289f21abb719aebc4/src/ResourceComparator.php#L43 actually does a strict comparison (resources are converted to a unique identifier, so aren't being compared on their values).

Reply:
Opened https://github.com/sebastianbergmann/comparator/pull/65.

Reply:
Explicitly reading the streams and compare the resulting strings would work here? Which would also cover un-seekable streams like `fopen('https://s3.amazonaws.com/bucket/file.xml')`